src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -1064,7 +1064,12 @@ void ClientHistory::trim_sync_history()
 bool ClientHistory::no_pending_local_changes(version_type version) const
 {
     ensure_updated(version);
-    for (size_t i = 0; i < sync_history_size(); i++) {
+    size_t base_version = 0;
+    auto upload_client_version =
+        version_type(m_arrays->root.get_as_ref_or_tagged(s_progress_upload_client_version_iip).get_as_int());
+    if (upload_client_version > m_sync_history_base_version)
```

Comment:
nit: size_t base_version = std::max(size_t(upload_client_version - m_sync_history_base_version), 0);

Reply:
version_type is unsigned so upload_client_version - m_sync_history_base_version will never be smaller than 0.